presto-parser/src/main/java/com/facebook/presto/sql/parser/ParsingWarning.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.parser;
+
+import com.facebook.presto.sql.tree.NodeLocation;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class ParsingWarning
+{
+    // Avoid extending ParsingException due to cost of filling in a stacktrace on creation
+    private final int line;
```

Comment:
Why not call these `lineNumber` and `columnNumber` similar to `NodeLocation`?

Reply:
It looks like NodeLocation uses line and charPositionInLine as the variable names even though the getters are named differently.
Would it make sense to update the field names for ParsingWarning, ParsingException and NodeLocation to match the getters in a separate commit/pull request?

Reply:
I agree with @nezihyigitbasi. We should call it `lineNumber` and `columnNumber` and avoid any kind of `+1` and `-1` magic